bin/ci

```diff
@@ -0,0 +1,109 @@
+source "$(dirname "${BASH_SOURCE[0]}")/_config.sh"
+
+function usage_ci() {
+    echo -e "Usage: ${1} [OPTIONS] COMMAND
+
+OPTIONS:
+  -h, --help   print this message
+
+COMMAND:
+  check_configuration  check that the \".circleci/config.yml\" file is up-to-date
+  update               generate the \".circleci/config.yml\" file for
+                       the current diff as compared to the master branch
+"
+}
+
+
+function check_changes() {
+    git whatchanged --name-only --pretty="" master | grep "${1}" &> /dev/null
+}
+
+
+# Utility function that updates circle configuration source files with a
+# workflow for each site and running only the necessary jobs
+#
+# Usage: update_sources
+function update_sources() {
+    # Reset the content of the .circleci/src/@root.yml file
+    disclaimer="# /!\ DO NOT EDIT: this file is autogenerated"
+    printf "%s\nparameters:\n" "${disclaimer}" > ".circleci/src/@parameters.yml"
+
+    while IFS= read -r -d '' site_path
+    do
+        # shellcheck disable=SC2001
+        site=$(echo "${site_path}" | sed 's|'"${SITES_DIRECTORY}\/"'||')
+
+        # Add a workflow for the site from our template
+        printf "%s\n" "${disclaimer}" > ".circleci/src/workflows/@${site}.yml"
+
+        # shellcheck disable=SC2016
+        sed 's|${SITE}|'"${site}"'|g' < .circleci/src/workflows/site.yml.tpl \
+            >> ".circleci/src/workflows/@${site}.yml"
+
+        # Conditions to run a site's workflow
+        if \
+            check_changes "^Dockerfile$" || \
+            check_changes "^${site_path}/VERSION$" || \
+            check_changes "^${site_path}/requirements/" || \
+            check_changes "^${site_path}/src/" ;
+        then
+            printf "    run_%s:\n        type: boolean\n        default: true\n" "${site}" \
+                >> ".circleci/src/@parameters.yml"
+        fi
+    done <  <(find "sites" -maxdepth 1 -mindepth 1 -type d -print0 | sort -z)
+}
+
+
+# Utility function that generates a circle configuration file with a workflow
+# for each site and running only the necessary jobs
+#
+# Usage: update
+function update() {
+    update_sources
+
+    # Generate the `.circleci/config.yml` file from the yaml source tree
+    docker run --rm -v "${PROJECT_DIRECTORY}:/data" circleci/circleci-cli:alpine \
+        config pack /data/.circleci/src > .circleci/config.yml
+
+    # Check config validity
+    docker run --rm -v "${PROJECT_DIRECTORY}:/data" circleci/circleci-cli:alpine \
+        config validate /data/.circleci/config.yml
+}
+
+
+# Utility function that checks if the circle configuration committed to git
+# corresponds to the state of the project so that the workflows and jobs are
+# run according to what has changed as compred to the "master" branch.
+#
+# Usage: check_configuration
+function check_configuration() {
+    update_sources
+    if \
+        git ls-tree master --name-only | grep .circleci/config.yml && \
+        git diff --name-only | grep "^.circleci/config.yml$";
```

Comment:
Alternative implementation:

```suggestion
        git ls-tree master --name-only | grep "^.circleci/src" && \
        ! git ls-tree master --name-only | grep "^.circleci/config.yml$";
```

Reply:
In fact the first line is only here to avoid failing on the present PR where `.circleci/config.yml` does not exist yet.

I think we can fun `bin/ci update` and check that nothing has changed in the `.circleci` folder...

Reply:
@jmaupetit I think I nailed it but take a look :pray: